cli/sessionstore/sessionstore_test.go

```diff
@@ -0,0 +1,156 @@
+package sessionstore_test
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"path"
+	"runtime"
+	"testing"
+
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/v2/cli/config"
+	"github.com/coder/coder/v2/cli/sessionstore"
+)
+
+type mockKeyringProvider struct {
+	err error
+}
+
+func (u mockKeyringProvider) Set(_, _, _ string) error { return u.err }
+
+func (u mockKeyringProvider) Get(_, _ string) (string, error) { return "", u.err }
+
+func (u mockKeyringProvider) Delete(_, _ string) error { return u.err }
+
+func TestOSKeyringWithFileFallback_Keyring(t *testing.T) {
+	if runtime.GOOS == "linux" {
+		t.Skip("this test isn't supported on Linux due to lack of a native keyring")
```

Comment:
By "native" I think you mean "not installed/configured by default" right?  There are keyrings like GNOME keyring, should we test with that?  It does require a dbus session though...might not be worth the effort lol